src/Conversion/ONNXToKrnl/Sequence/SequenceInsert.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------SequenceInsert.cpp - Lowering SequenceInsert Op---------------=== //
+//
+// Copyright 2020-2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the ONNX SequenceInsert Operator to Krnl dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToKrnl/ONNXToKrnlCommon.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+
+using namespace mlir;
+
+struct ONNXSequenceInsertOpLowering : public ConversionPattern {
+  ONNXSequenceInsertOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(typeConverter,
+            mlir::ONNXSequenceInsertOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op->getLoc();
+    ONNXSequenceInsertOpAdaptor operandAdaptor(operands);
+    ONNXSequenceInsertOp thisOp = dyn_cast<ONNXSequenceInsertOp>(op);
+    MultiDialectBuilder<KrnlBuilder, MathBuilder> create(rewriter, loc);
+    IndexExprScope IEScope(&rewriter, loc);
+
+    // ONNXSequenceInsertOp sequenceInsertOp = cast<ONNXSequenceInsertOp>(op);
+
+    // Get element type for seq from the output
+
+    auto input_sequence = operandAdaptor.input_sequence();
+    auto dimSize = rewriter.create<memref::DimOp>(
+        loc, input_sequence, create.math.constantIndex(0));
+    SymbolIndexExpr boundIE(dimSize);
+
+    // Can not use this type because the output element type may be more precise
+    // This issue may be resolved in shape inference by backwards propagation
+    // In some sense, the type info for seq is not consistent at SequenceInsert
+    // The code of type converter for Seq is duplicated
+    // auto seqElementType =
+    // input_sequence.getType().cast<MemRefType>().getElementType();
+
+    ShapedType seqElementType =
+        thisOp.getResult().getType().cast<SeqType>().getElementType();
+    Type elementType = seqElementType.getElementType();
+    Type seqElementConvertedType;
+    if (seqElementType.hasRank()) {
```

Comment:
[Suggestion] use ternary operator (?:) to initialize `seqElementConvertedType` ?